syntax = "proto3";

package relay_node;

message FromRelayNodeRequest {
    repeated double values = 1;
    int32 player_id = 2;
    bool shoot_detected = 3;
    bool ir_detected = 4;
    optional string test_action = 5;
}

message Player {
    int32 hp = 1;
    int32 bullets = 2;
    int32 bombs = 3;
    int32 shield_hp = 4;
    int32 deaths = 5;
    int32 shields = 6;
}

message FromRelayNodeResponse {
    Player player_one = 1;
    Player player_two = 2;
}

service RelayNode {
    rpc processAi(FromRelayNodeRequest) returns (FromRelayNodeResponse);
}
